src/Views/Appearance.vala

```diff
@@ -76,6 +92,51 @@ public class Appearance : Gtk.Grid {
 
         update_text_size_modebutton (interface_settings);
 
+        Pantheon.AccountsService? pantheon_act = null;
+        FDO.Accounts? accounts_service = null;
+        string? user_path = null;
+        try {
+            accounts_service = GLib.Bus.get_proxy_sync (
+                GLib.BusType.SYSTEM,
+               "org.freedesktop.Accounts",
+               "/org/freedesktop/Accounts"
+            );
+
+            user_path = accounts_service.find_user_by_name (GLib.Environment.get_user_name ());
+        } catch (Error e) {
+            critical (e.message);
+        }
+
+        try {
+            pantheon_act = GLib.Bus.get_proxy_sync (
+                GLib.BusType.SYSTEM,
+                "org.freedesktop.Accounts",
+                user_path,
+                GLib.DBusProxyFlags.GET_INVALIDATED_PROPERTIES
+            );
+        } catch (Error e) {
+            warning ("Unable to get AccountsService proxy, color scheme preference may be incorrect");
+        }
+
+        // FIXME: This seems… not ideal. Can't we bind this?
+        switch (pantheon_act.prefers_color_scheme) {
```

Comment:
Connecting to this DBus interface can fail which is handled by the catch above. However, this here will still execute. Check for `pantheon_act != null` before this switch.